### DISCUSSION TITLE
Merge a branch 500 error

### DISCUSSION THREAD

**Discussion by minsoo220** [community/community#70192] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I called the [api](https://docs.github.com/ko/rest/branches/branches?apiVersion=2022-11-28#merge-a-branch) using github cli, cURL, and JavaScript, but all failed with error 500.

Can I know the exact error details and why it fails?

Request
```
* Request at 2023-10-16 12:31:54.957271 +0900 KST m=+0.071710459
* Request to https://api.github.com/repos/OWNER/REPO/merges
> POST /repos/OWNER/REPO/merges HTTP/1.1
> Host: api.github.com
> Accept: application/vnd.github+json
> Authorization: token ~~
> Content-Length: 37
> Content-Type: application/json; charset=utf-8
> Time-Zone: Asia/Seoul
> User-Agent: GitHub CLI 2.36.0
> X-Github-Api-Version: 2022-11-28

{
  "base": "release-787c",
  "head": "main"
}
```

Response
```
< HTTP/2.0 500 Internal Server Error
< Access-Control-Allow-Origin: *
< Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
< Content-Length: 0
< Content-Security-Policy: default-src 'none'
< Content-Type: application/json; charset=utf-8
< Date: Mon, 16 Oct 2023 03:31:57 GMT
< Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
< Server: GitHub.com
< Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
< Vary: Accept-Encoding, Accept, X-Requested-With
< X-Accepted-Oauth-Scopes:
< X-Content-Type-Options: nosniff
< X-Frame-Options: deny
< X-Github-Api-Version-Selected: 2022-11-28
< X-Github-Media-Type: github.v3; format=json
< X-Github-Request-Id: DF51:348C8:80BD87:984D2C:652CAEAB
< X-Oauth-Client-Id: oauth client id
< X-Oauth-Scopes: gist, read:org, repo, workflow
< X-Ratelimit-Limit: 5000
< X-Ratelimit-Remaining: 4996
< X-Ratelimit-Reset: 1697430591
< X-Ratelimit-Resource: core
< X-Ratelimit-Used: 4
< X-Xss-Protection: 0
```

**Comment by seyLu** [comment]
#### Hi @minsoo220,

500 server error means that there is an error on the server's side (in this case, Github), and not on the client side (in this case, your machine). Since the error came from the server side, the client can't really do anything about it rather than wait until the issue on the server side gets resolved.

As for the listing out PR merges, there is currently no `merges` endpoint as per the latest Github API version `2022-11-28`. You should be getting status 404 instead of 500 if there were no server errors.

As a workaround, you can use the `search` endpoint instead:  
```bash
https://api.github.com/search/issues?q=repo:{OWNER}/{REPO}+is:merged
```

You can read more on:
- [MDN docs: status 500](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500)
- [Github docs: Search based on whether a pull request has been merged](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-based-on-whether-a-pull-request-is-merged-or-unmerged)

**Reply by minsoo220** [reply]
@seyLu 
Thanks for the comment

However, the error occurred in the branch merge API, not the merges pr list API.
<img width="1257" alt="image" src="https://github.com/community/community/assets/118704536/02ef1cbc-1362-4dae-a4ea-2667528a57fd">

The API I used is from the latest document.

This is the English document URL
https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#merge-a-branch

**Comment by ghostinhershell** [comment]
Hi there @minsoo220, 

Thanks so much for reaching out in the community. As @seyLu mentioned above, 500 server error typically indicate that there's an error on GitHub's side. For future reference, if you do run into a 500 error when working with GitHub, check out our status page at https://www.githubstatus.com/. 

Are you still seeing this error on your end when attempting to merge a branch?

**Reply by minsoo220** [reply]
@ghostinhershell  Yes, the code below is an error that occurred an hour ago.

```
RequestError [HttpError]: invalid json response body at https://api.github.com/repos/[owner]/[repo]/merges reason: Unexpected end of JSON input
branch.name release-9ced
    at /home/runner/work/_actions/actions/github-script/v6/dist/index.js:6864:11
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async module.exports (/home/runner/work/[repo]/[repo]/.github/update-branch.js:43:7) {
  status: 500,
  request: {
    method: 'POST',
    url: 'https://api.github.com/repos/[owner]/[repo]/merges',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'actions/github-script octokit-core.js/3.6.0 Node.js/16.20.2 (linux; x64)',
      authorization: 'token [REDACTED]',
      'content-type': 'application/json; charset=utf-8'
    },
    body: '{"base":"release-2e87","head":"main"}',
    request: { agent: [Agent], hook: [Function: bound bound register] }
  }
}
```

**Reply by jayrifkinsv** [reply]
I'm seeing the same 500 error with my repo from the `/repos/OWNER/REPO/merges` endpoint, trying to merge one branch into another.  
https://www.githubstatus.com/ shows all statuses as Normal
